modules/memfault/memfault_integration.c

```diff
@@ -90,11 +90,10 @@ void memfault_platform_get_device_info(sMemfaultDeviceInfo *info)
 
 static int request_imei(const char *cmd, char *buf, size_t buf_len)
 {
-	enum at_cmd_state at_state;
-	int err = at_cmd_write(cmd, buf, buf_len, &at_state);
+	int err = nrf_modem_at_cmd(buf, buf_len, cmd);
```

Comment:
Please mark that `nrf_modem_at_cmd` also will return positive errors on modem error, +CME error and +CMS error. Suggestion is to log positive retval of `nrf_modem_at_printf` before returning -ENOEXEC.